sql/parser/sql.y

```diff
@@ -1697,15 +1696,30 @@ opt_table:
 | /* EMPTY */ {}
 
 all_or_distinct:
-  ALL {}
-| DISTINCT {}
-| /* EMPTY */ {}
+  ALL
+  {
+    $$ = true
+  }
+| DISTINCT
+  {
+    $$ = false
+  }
+| /* EMPTY */
+  {
+    $$ = false
+  }
 
 // We use (NIL) as a placeholder to indicate that all target expressions should
```

Comment:
This comment is now inaccurate. I think it is a left-over from the postgres grammar. You should update it to say that we use a non-nil empty list as a placeholder.